File: src/utils/formatValue.ts
import type { Entry, TrackType } from "../types";

export function getValueInputLabel(trackType: TrackType): string {
  if (trackType.valueType === "duration") {
    const unit = trackType.durationUnit === "hours" ? "hours" : "minutes";
    return `Duration (${unit})`;
  }
  if (trackType.valueType === "count") {
    return trackType.valueUnit
      ? `Count (${trackType.valueUnit})`
      : "Count";
  }
  return "Value";
}

export function formatEntryValue(entry: Entry, trackType: TrackType): string | null {
  if (entry.value == null) return null;
  if (trackType.valueType === "duration") {
    const unit = trackType.durationUnit === "hours" ? "hr" : "min";
    return `${entry.value} ${unit}`;
  }
  if (trackType.valueType === "count" && trackType.valueUnit) {
    return `${entry.value} ${trackType.valueUnit}`;
  }
  return `${entry.value}`;
}

File: life-tracker-ios/life-tracker-ios/Helpers/FormatValue.swift
//
//  FormatValue.swift
//  life-tracker-ios
//

import Foundation

enum FormatValue {
    /// Format entry value for display (e.g. "5", "30 min", "1.5")
    static func formatEntryValue(_ entry: Entry, trackType: TrackType?) -> String? {
        guard let v = entry.value else { return nil }
        let unit = trackType?.valueUnit ?? trackType?.metadata?["unit"] ?? ""
        if unit.isEmpty { return String(format: "%g", v) }
        return "\(String(format: "%g", v)) \(unit)"
    }
}
